ietf/static/js/agenda_timezone.js

```diff
@@ -8,272 +7,300 @@
+                        .html(format_time(this.start_ts, newtz, this.format) + '<span class="d-lg-none"><br></span>-' +
+                            format_time(this.end_ts, newtz, this.format));
+                }
+            });
+        update_tooltips_all();
+        update_clock();
+    }
+
+    // Update hrefs in anchor tags with the "now-link" class. Mark the target with the "current-session" class.
+    function update_now_link(agenda_rows, ongoing_rows, later_rows) {
+        agenda_rows.removeClass('current-session');
+        const links_to_update = $('a.now-link');
+        if (ongoing_rows.length > 0) {
+            // sessions are ongoing - find those with the latest start time and mark the first of them as "now"
+            const last_start_time = ongoing_rows[ongoing_rows.length - 1].slot_start_ts;
+            for (let ii=0; ii < ongoing_rows.length; ii++) {
+                const dt = ongoing_rows[ii].slot_start_ts.diff(last_start_time, 'seconds');
+                if (Math.abs(dt) < 1) {
+                    $(ongoing_rows[ii]).addClass('current-session');
+                    links_to_update.attr('href', '#' + ongoing_rows[ii].id);
+                    break;
                 }
-            } else {
-                $(this)
-                    .html(format_time(this.start_ts, newtz, this.format) + '<span class="d-lg-none"><br></span>-' +
-                        format_time(this.end_ts, newtz, this.format));
             }
-        });
-    update_tooltips_all();
-    update_clock();
-};
+        } else if (later_rows.length > 0) {
+            // There were no ongoing sessions, look for the next one to start and mark as current
+            $(later_rows[0]).addClass('current-session');
+            links_to_update.attr('href', '#' + later_rows[0].id);
+        } else {
+            // No sessions in the future - meeting has apparently ended
+            links_to_update.attr('href', '#');
+            links_to_update.addClass('text-muted text-decoration-line-through'); // mark link
+        }
+    }
 
-// Highlight ongoing based on the current time
-window.highlight_ongoing = function () {
-    $("div#now")
-        .remove("#now");
-    $('.table-warning')
-        .removeClass("table-warning");
-    var agenda_rows = $('[data-slot-start-ts]');
-    agenda_rows = agenda_rows.filter(function () {
-        return moment()
-            .isBetween(this.slot_start_ts, this.slot_end_ts);
-    });
-    agenda_rows.addClass("table-warning");
-    agenda_rows.first()
-        .children("th, td")
-        .prepend($('<div id="now"></div>'));
-};
+    function update_ongoing_sessions() {
+        const agenda_rows = $('[data-slot-start-ts]');
+        const ongoing_rows = agenda_rows.filter(function () {
+            return moment()
+                .isBetween(this.slot_start_ts, this.slot_end_ts);
+        });
+        const later_rows = agenda_rows.filter(function() { return moment().isBefore(this.slot_start_ts); });
```

Comment:
There's a small race condition here, since `moment()` is executed multiple times.  You could end up with a row being left out of `ongoing_rows` and `later_rows`.

Reply:
Good catch, thanks